api/services/gameService.js

```diff
@@ -38,9 +38,10 @@ function tempUser(usr, points) {
   this.faceCards.sort(comapreByRankThenSuit);
 }
 
-function tempGame(game, p0, p1) {
+function tempGame(game, p0, p1, spectatingUsers) {
```

Comment:
since `spectatingUsers` is already on the `game` object when we populate that collection, let's remove it from here as an argument and instead assign the value of `this.spectators` to the `.map()` call that you're doing below;

```
this.spectatingUsers = game.spectatingUsers.map((user) => user.username);
```